cupyx/scipy/ndimage/_spline_prefilter_core.py

```diff
@@ -169,7 +169,9 @@ def _get_spline1d_code(mode, poles, n_boundary):
         for (i = 1; i < n; ++i) {{
             c[i * element_stride] += z * c[(i - 1) * element_stride];
         }}''')
-
+        code.append('''
+        __syncthreads();
+        ''')
```

Comment:
should this have an ifdef here as in #4504?

```suggestion
        code.append('''
        #ifdef __HIP_DEVICE_COMPILE__
        __syncthreads();
        #endif
        ''')
```

Reply:
Thank you!